src/Microsoft.SqlTools.ServiceLayer/Connection/ConnectionService.cs

```diff
@@ -1159,18 +1163,24 @@ private void RunConnectRequestHandlerTask(ConnectParams connectParams)
             catch (Exception ex)
             {
                 newResponse.Result = false;
-                newResponse.ErrorMessage = ex.InnerException != null ? (ex.Message + Environment.NewLine + Environment.NewLine + ex.InnerException.Message) : ex.Message;
-                newResponse.ErrorMessage = Regex.Replace(newResponse.ErrorMessage, @"\r?\nChanged database context to '\w+'\.", "");
-                newResponse.ErrorMessage = Regex.Replace(newResponse.ErrorMessage, @"\r?\nChanged language setting to \w+\.", "");
-                if (newResponse.ErrorMessage.Equals(SR.PasswordChangeEmptyPassword))
+                newResponse.ErrorMessage = ex.Message;
+                int errorCode = 0;
+                SqlError endError = (ex.InnerException as SqlException)?.Errors[0];

```

Comment:
Would recommend checking if 'Errors' is not empty too, just to avoid any possible NREs